pkgs/development/libraries/java/swt/default.nix

```diff
@@ -1,43 +1,57 @@
-{ lib, stdenv, fetchurl, canonicalize-jars-hook
-, gtk2, jdk, libGL, libGLU, libXi, libXt, libXtst, libsoup, webkitgtk
-, gdk-pixbuf, glib, pango, xorg
-, unzip, pkg-config
+{ lib, stdenv, fetchzip, pkgsBuildHost
+, atk, glib, gtk2, jdk, libGL, libGLU, libXi, libXt, libXtst, libsoup
+, webkitgtk, xorg
 }:
 
 let
   platformMap = {
     x86_64-linux =
       { platform = "gtk-linux-x86_64";
-        sha256 = "1qq0pjll6030v4ml0hifcaaik7sx3fl7ghybfdw95vsvxafwp2ff"; };
+        sha256 = "17frac2nsx22hfa72264as31rn35hfh9gfgy0n6wvc3knl5d2716"; };
     i686-linux =
       { platform = "gtk-linux-x86";
-        sha256 = "03mhzraikcs4fsz7d3h5af9pw1bbcfd6dglsvbk2ciwimy9zj30q"; };
+        sha256 = "13ca17rga9yvdshqvh0sfzarmdcl4wv4pid0ls7v35v4844zbc8b"; };
     x86_64-darwin =
       { platform = "cocoa-macosx-x86_64";
-        sha256 = "00k1mfbncvyh8klgmk0891w8jwnd5niqb16j1j8yacrm2smmlb05"; };
+        sha256 = "0wjyxlw7i9zd2m8syd6k1q85fj8pzhxlfsrl8fpgsj37p698bd0a"; };
   };
 
-  metadata = assert platformMap ? ${stdenv.hostPlatform.system}; platformMap.${stdenv.hostPlatform.system};
-
+  metadata = assert platformMap ? ${stdenv.hostPlatform.system};
+    platformMap.${stdenv.hostPlatform.system};
 in stdenv.mkDerivation rec {
+  pname = "swt";
   version = "4.5";
   fullVersion = "${version}-201506032000";
-  pname = "swt";
 
   hardeningDisable = [ "format" ];
 
   # Alas, the Eclipse Project apparently doesn't produce source-only
   # releases of SWT.  So we just grab a binary release and extract
   # "src.zip" from that.
-  src = fetchurl {
-    url = "http://archive.eclipse.org/eclipse/downloads/drops4/R-${fullVersion}/${pname}-${version}-${metadata.platform}.zip";
+  src = fetchzip {
+    url = "https://archive.eclipse.org/eclipse/downloads/drops4/" +
+      "R-${fullVersion}/${pname}-${version}-${metadata.platform}.zip";
     sha256 = metadata.sha256;
-  };
+    stripRoot = false;
+    extraPostFetch = ''
+      mkdir "$unpackDir"
+      cd "$unpackDir"
 
-  sourceRoot = ".";
+      renamed="$TMPDIR/src.zip"
+      mv "$out/src.zip" "$renamed"
+      unpackFile "$renamed"
+      rm -r "$out"
 
-  nativeBuildInputs = [ canonicalize-jars-hook unzip pkg-config ];
+      mv "$unpackDir" "$out"
+    '';
+  };
```

Comment:
Why is this needed? Maybe it should be commented?

Reply:
This `extraPostFetch` performs the second unpack necessary.

Reply:
> This `extraPostFetch` performs the second unpack necessary.

You mean upstream sends us a zip inside a zip?

Reply:
Yeah. Per the comment above:

```nix
  # Alas, the Eclipse Project apparently doesn't produce source-only
  # releases of SWT.  So we just grab a binary release and extract
  # "src.zip" from that.
```